datafusion/core/src/physical_optimizer/test_utils.rs

```diff
@@ -361,3 +364,17 @@ pub fn sort_exec(
     let sort_exprs = sort_exprs.into_iter().collect();
     Arc::new(SortExec::new(sort_exprs, input))
 }
+
+pub fn check_integrity<T: Clone>(context: PlanContext<T>) -> Result<PlanContext<T>> {
+    context.transform_up(&|node| {
+        let children_plans = node.plan.children();
+        assert_eq!(node.children.len(), children_plans.len());
+        for (child_plan, child_node) in children_plans.iter().zip(node.children.iter()) {
+            assert_eq!(
```

Comment:
Can you instead simply compare the plans directly rather than converting them to strings?

Reply:
Since `ExecutionPlan`'s do not implement `PartialEq`, we lack a straightforward way to compare them. However, I believe there are no obstacles to implementing `PartialEq` support for `ExecutionPlan`'s.

Reply:
Makes sense -- maybe that is something that can be embedded in a comment to help anyone else that may have the same question